lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -40,8 +40,7 @@ defmodule LambdaEthereumConsensus.ForkChoice do
 
   @spec has_block?(Types.root()) :: boolean()
   def has_block?(block_root) do
-    block = get_block(block_root)
-    block != nil
+    GenServer.call(__MODULE__, {:has_block?, block_root}, @default_timeout)
```

Comment:
I had to make this change because the tree lives only on memory. We can improve this by:
1. persist metadata about blocks (currently we only have implicit info from the tree)
2. load previous state when restarting the node (with metadata)
3. differentiate between `has_block_in_db?`, `can_process_block?`, and `block_was_processed?`

Reply:
can you create a ticket?